added support for markdown.nvim | changed some native markup/down highlights TBD if final

avih7531 opened the pull request:
Added support for markdown.nvim

Had to change multiple `@markup` highlight groups because of preset bolds and italics, to get a nice looking `RenderMarkdown`, but think it came out great. 

Any thoughts on whether or not to keep the bolds for H1-3 and italics for H5-6 permanently? If so, should they persist in the `RenderMarkdown` groups? My vote would be no as I'm unfamiliar with the convention for highlights like that, but then again, not my repo :)

AH

avih7531 commented:
![08-04-2024_19:51:56 253](https://github.com/user-attachments/assets/42f32507-6577-416d-9094-a8402610c9f7)
![08-04-2024_19:52:11 537](https://github.com/user-attachments/assets/1fb79803-4d7a-48fa-a83a-effceba097be)
![08-04-2024_19:52:28 011](https://github.com/user-attachments/assets/581c9003-bb46-4382-bd94-e9e8d03127b3)

5-pebbles commented on lua/nordic/groups/integrations.lua:
```diff
@@ -523,6 +524,24 @@ function M.get_groups()
+    G.RenderMarkdownH4 = { link = '@markup.heading.4' }
+    G.RenderMarkdownH5 = { link = '@markup.heading.5' }
+    G.RenderMarkdownH6 = { link = '@markup.heading.6' }
+    G.RenderMarkdownH1Bg = { bg = U.blend(C.yellow.base, C.grey0, 0.3) }
+    G.RenderMarkdownH2Bg = { bg = U.blend(C.orange.base, C.grey0, 0.3) }
+    G.RenderMarkdownH3Bg = { bg = U.blend(C.magenta.base, C.grey0, 0.3) }
+    G.RenderMarkdownH4Bg = { bg = U.blend(C.green.base, C.grey0, 0.3) }
+    G.RenderMarkdownH5Bg = { bg = U.blend(C.blue2, C.grey0, 0.3) }
+    G.RenderMarkdownH6Bg = { bg = U.blend(C.cyan.base, C.grey0, 0.3) }
```
This should blend with `C.bg` because we have multiple background colors and the user can specify their own.

```suggestion
    G.RenderMarkdownH1Bg = { bg = U.blend(C.yellow.base, C.bg, 0.3) }
    G.RenderMarkdownH2Bg = { bg = U.blend(C.orange.base, C.bg, 0.3) }
    G.RenderMarkdownH3Bg = { bg = U.blend(C.magenta.base, C.bg, 0.3) }
    G.RenderMarkdownH4Bg = { bg = U.blend(C.green.base, C.bg, 0.3) }
    G.RenderMarkdownH5Bg = { bg = U.blend(C.blue2, C.bg, 0.3) }
    G.RenderMarkdownH6Bg = { bg = U.blend(C.cyan.base, C.bg, 0.3) }
```

Also I think the blend strength should be lower (0.15) but that's just personal preference.

avih7531 replied:
Sounds good. I 'stole' the .3 from cattpuuchin's implementation, but agree .15 looks better.

5-pebbles commented on lua/nordic/groups/integrations.lua:
```diff
@@ -348,17 +349,17 @@ function M.get_groups()
-    G['@markup.heading.3'] = { fg = C.magenta.base, bold = true }
+    G['@markup.heading.1'] = { fg = C.yellow.base }
+    G['@markup.heading.2'] = { fg = C.orange.base }
+    G['@markup.heading.3'] = { fg = C.magenta.base }
     G['@markup.heading.4'] = { fg = C.green.base }
-    G['@markup.heading.5'] = { fg = C.blue2, italic = true }
-    G['@markup.heading.6'] = { fg = C.cyan.base, italic = true }
-    G['@markup.italic'] = { italic = true }
+    G['@markup.heading.5'] = { fg = C.blue2 }
+    G['@markup.heading.6'] = { fg = C.cyan.base }
```
What about making all headings bold? I like that best, and that's what [tokyonight](https://github.com/folke/tokyonight.nvim) does.

I know the old bold and italic system is what GitHub uses, but when I make a heading, I am just trying to add another layer of structure, not make things italic.

5-pebbles commented on lua/nordic/groups/integrations.lua:
```diff
@@ -348,17 +349,17 @@ function M.get_groups()
     --- Markup
     G['@markup'] = { link = '@none' }
-    G['@markup.emphasis'] = { italic = true }
+    G['@markup.emphasis'] = { fg = C.orange.base, italic = true }
```
I see why you did this, and I think it makes it easier to edit the rendering (by making those things noticeable). However, I think it should just occur on rendered text because it is distracting otherwise.

avih7531 replied:
Fair enough. That might just be a personal preference, but I noticed it adds so much to a markdown if emphasis, bold, and italic are different colors

5-pebbles commented on lua/nordic/groups/integrations.lua:
```diff
@@ -348,17 +349,17 @@ function M.get_groups()
-    G['@markup.italic'] = { italic = true }
+    G['@markup.heading.5'] = { fg = C.blue2 }
+    G['@markup.heading.6'] = { fg = C.cyan.base }
+    G['@markup.italic'] = { fg = C.orange.base, italic = true }
```
same as `@markup.emphasis`

5-pebbles commented on lua/nordic/groups/integrations.lua:
```diff
@@ -370,7 +371,7 @@ function M.get_groups()
     G['@markup.math'] = { link = 'Special' }
     G['@markup.raw'] = { link = 'String' }
     G['@markup.raw.markdown_inline'] = { bg = C.black2, fg = C.fg }
-    G['@markup.strong'] = { bold = true }
+    G['@markup.strong'] = { fg = C.orange.base, bold = true }
```
same as `@markup.emphasis`

5-pebbles commented on lua/nordic/groups/native.lua:
```diff
@@ -74,8 +74,8 @@ function M.get_groups()
     G.markdownCodeBlock = { link = 'CodeBlock' }
-    G.markdownH1 = { link = 'htmlH1' }
-    G.markdownH2 = { link = 'htmlH2' }
+    G.markdownH1 = { fg = C.yellow.base }
+    G.markdownH2 = { fg = C.orange.base }
```
I don't know what this does could you explain it?

avih7531 replied:
Unfortunately, to get the headings to work, both the `@markup.heading.x` groups and the `markdownHx` groups had to be changed in order to get the desired settings to work. I unlinked those because it would make `H1-2` bold, but if we decide to make `H1-6` bold, we can add that link back

5-pebbles reviewed (changes requested):
Thanks for the screenshots. I should note this is not my repository either, the following review is just my opinion.

avih7531 commented:
Appreciate the code review, by the way -- I've never had one. Will get back later today with a revised version

5-pebbles commented:
Alright, make commits for bold headers (with relinking markdownH1 + markdownH2) and the background blend. I would leave the orange emphasis for Alex to look at, though.

5-pebbles commented:
Here are some examples of the orange and not orange markdown section for comparison:

![image](https://github.com/user-attachments/assets/c3f3a524-9af9-4df6-8f3a-8d88745f0739)
![image](https://github.com/user-attachments/assets/9a8c3997-a18b-4209-81e0-b71876b63133)

![image](https://github.com/user-attachments/assets/7a898aa9-c9c3-40a7-a4fc-30c82cdb8eb0)
![image](https://github.com/user-attachments/assets/deb34b12-5fb1-4af8-880f-2bbc376b030e)

and with rendering:
![image](https://github.com/user-attachments/assets/242d40d5-3e3a-44d9-8413-ee91cadd9ba3)
![image](https://github.com/user-attachments/assets/18a4cfa1-8782-4b7f-9904-900e9593297a)

avih7531 commented:
Just committed changes to 'avih7531:nordic.nvim' hopefully that updated here -- sorry not so good with github!!

5-pebbles commented:
> Just committed changes to 'avih7531:nordic.nvim' hopefully that updated here -- sorry not so good with github!!

I am assuming you are talking about the one three days ago: 381b17c If not, you still need to push your changes.

I had never worked with github before helping with this repo, so don't worry about inexperience. Also, you should be able to see all commits on the `Commits` tab of this page (at the top) and all changes in the `Files changed` tab.

avih7531 commented:
Yes! Commit [381b17c](https://github.com/AlexvZyl/nordic.nvim/commit/381b17c53ed268a75a3b04884448c27846586f6f)  should be what was asked before hand. Changed all headings to bold, re-linked `markdownH1&2`, and fixed the `C.bg` blend. Left italics and bolds as is for Alex to decide.

Appreciate your patience and support working with Git! Definitely new to me, but learning.

AlexvZyl commented:
@5-pebbles Please resolve conversations that you are happy with :)

5-pebbles commented:
Alright, @avih7531 I am going to ask that you revert the orange emphasis. I have been using it since this **PR** was opened, and I still find it a little distracting. It's also fairly easy to add with the new `on_highlight` option.

avih7531 commented:
Thanks both! Will have that done by tonight -- appreciate giving me some git practice, too

AlexvZyl commented:
Ping @avih7531

5-pebbles commented:
@avih7531 (•ᴥ• )́ If you would like I could finish this pr...

AlexvZyl commented:
This will have to get added in a different release.

AlexvZyl commented:
@avih7531 Bump.

Otherwise I am closing this as stale.

AlexvZyl commented:
Closing as stale.  @avih7531 you are more than welcome to re-open this PR.

AlexvZyl commented:
Closing as stale @avih7531.  You are more than welcome to re-open.
